### ISSUE TITLE
Cleaner implmentation for VoiceClient

### ISSUE THREAD

**Issue by slorello89** [Vonage/vonage-dotnet-sdk#221]
As it stands right now, the [IVoiceClient](https://github.com/Vonage/vonage-dotnet-sdk/blob/master/Vonage/Voice/IVoiceClient.cs#L15) is a direct wrapper around our [Voice REST API](https://developer.nexmo.com/api/voice#createCall). This is usable but a tad clunky as it requires the user to create a To/From endpoint object, and either provide an answer URL or NCCO Object. 
Right now, creating a call looks like [this](https://developer.nexmo.com/voice/voice-api/code-snippets/make-an-outbound-call-with-ncco):

```csharp
var toEndpoint = new PhoneEndpoint() { Number = TO_NUMBER };
var fromEndpoint = new PhoneEndpoint() { Number = NEXMO_NUMBER };

var talkAction = new TalkAction() { Text = "This is a text to speech call from Nexmo" };
var ncco = new Ncco(talkAction);

var command = new CallCommand() { To = new Endpoint[] { toEndpoint }, From = fromEndpoint, Ncco = ncco, EventUrl = EVENT_URL };
var response = client.VoiceClient.CreateCall(command);
```

It would be nice if we could add a couple of new APIs to this client so that requests could look something like this:

```csharp
var talkAction = new TalkAction() { Text = "This is a text to speech call from Nexmo" };
var ncco = new Ncco(talkAction);
var response = client.VoiceClient.CreateCall(toNumber, fromNumber, ncco);
```

Or in the case where you'd not be dialing out to a PSTN endpoint like:

```csharp
var talkAction = new TalkAction() { Text = "This is a text to speech call from Nexmo" };
var ncco = new Ncco(talkAction);
var response = client.VoiceClient.CreateCall(toEndpoint, fromNumber, ncco);
```

**Comment by gagandeepp** [comment]
Interested , please assign this to me

**Comment by slorello89** [comment]
Feel free to take this @gagandeepp :)

**Comment by gagandeepp** [comment]
https://github.com/Vonage/vonage-dotnet-sdk/pull/231

**Comment by slorello89** [comment]
This issue is resolved by #231
